test/metabase/upload_test.clj

```diff
@@ -422,6 +442,29 @@
                       (t2/select-one Field :database_position 1 :table_id (:id table)))))
             (is (some? table))))))))
 
+(deftest load-from-csv-offset-datetime-test
+  (testing "Upload a CSV file with a datetime column"
+    (mt/test-drivers (mt/normal-drivers-with-feature :uploads)
+      (mt/with-empty-db
+        (with-mysql-local-infile-activated
+          (upload/load-from-csv!
+           driver/*driver*
+           (mt/id)
+           "upload_test"
+           (csv-file-with ["offset_datetime"
+                           "2022-01-01T01:00:00-07:00"])))
```

Comment:
I'm concerned the driver code will convert a GMT timezone to a string like "2022-01-01T01:00:00Z" with a "Z" at the end before inserting, and that it will not be a valid format for the database.

I tested this with MySQL and indeed it was the case:
```
DROP TABLE IF EXISTS timestamps;

CREATE TABLE timestamps (
    event_timestamp TIMESTAMP NOT NULL
);

INSERT INTO timestamps (event_timestamp) VALUES ('2022-01-01T01:00:00Z');

SELECT * from timestamps;
```

I'd suggest covering this corner case as a test here.

Reply:
Handled in [this commit](https://github.com/metabase/metabase/pull/34826/commits/8c3027f7cfbf1bc96a651e32977f407dffdf2da3)